### ISSUE TITLE
Validation for VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01912 is not correct according to the Vulkan spec.

### ISSUE THREAD

**Issue by matasx8** [KhronosGroup/Vulkan-ValidationLayers#5671]
**Environment:**
 - OS: Windows 10
 - GPU: RTX 3070ti
 - SDK or header version if building from repo: Vulkan SDK 1.3.243.0
 - Options enabled (synchronization, best practices, etc.):

**Describe the Issue**
VVL error introduced in this [PR](https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/5277/files#diff-4311fff80ea2686e702e6fb7fc1a33f8963ca268ea3212fe9ccae3a2434e60d9R590) is not correct according to the Vulkan spec. The spec says ([source](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdEndDebugUtilsLabelEXT.html)): 

> An application may open a debug label region in one command buffer and close it in another, or otherwise split debug label regions across multiple command buffers or multiple queue submissions. When viewed from the linear series of submissions to a single queue, the calls to [vkCmdBeginDebugUtilsLabelEXT](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdBeginDebugUtilsLabelEXT.html) and [vkCmdEndDebugUtilsLabelEXT](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdEndDebugUtilsLabelEXT.html) must be matched and balanced.

This validation check implemented [here](https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/5277/files#diff-8c4c0adae5e8a48e96ed1ad2569f5c065155cfca223e50cac4c13af0c7e0b5f2R1687) checks for matching debug labels *per command buffer* while it should check *per queue*. There's an exception for secondary command buffers ([source](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdEndDebugUtilsLabelEXT.html)):

> VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01913
If commandBuffer is a secondary command buffer, there must be an outstanding vkCmdBeginDebugUtilsLabelEXT command recorded to commandBuffer that has not previously been ended by a call to vkCmdEndDebugUtilsLabelEXT

In that case checking for matching debug labels makes sense _per secondary command buffer_.

**Steps to reproduce**
1. Download and install Unity 2022.2.4f1 or above (e.g. https://unity.com/releases/editor/whats-new/2022.2.15)
2. Install VVL that has this [PR](https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/5277/files#diff-4311fff80ea2686e702e6fb7fc1a33f8963ca268ea3212fe9ccae3a2434e60d9R590) landed
3. Launch Unity with any project
4. Switch Graphics API to Vulkan (https://docs.unity3d.com/Manual/GraphicsAPIs.html)
5. Launch Unity again with `--force-vulkan-layers` argument from CLI.
6. Observe `VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01912` get logged in the Unity console.

**Expected behavior**
VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01912 gets validated per queue
VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01913 gets validated per secondary command buffer

**Valid Usage ID**
VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01912

**Comment by akb825** [comment]
I'm seeing the same issue on Linux with my own engine. ([DeepSea](https://github.com/akb825/DeepSea)) I have debug labels that span multiple command buffers across queue submission that is causing this error to be reported every frame. According to the spec this usage should be valid:
> An application may open a debug label region in one command buffer and close it in another, or otherwise split debug label regions across multiple command buffers or multiple queue submissions.

**Comment by jimblandy** [comment]
@juan-lunarg Do you think this will be quick to fix? I had started looking into doing it myself today, but this is my first time looking at the code.

**Comment by juan-lunarg** [comment]
> Do you think this will be quick to fix? 

I think so. I'll start working on this.

**Comment by juan-lunarg** [comment]
I've created a test that reproduces the false positive. Should have a fix soon.

**Comment by juan-lunarg** [comment]
TLDR:

There is some spec confusion that needs ironing out.

For now we are going to remove validation for `VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01912` entirely since it's in the wrong place in the spec regardless of current confusion.

I'm also keeping the positive test I've written to prevent a regression.

**Comment by juan-lunarg** [comment]
False positive has been addressed and we have added testing to prevent a regression.

We will also need to work on some minor issues in the specification regarding this extension.

Thank you for bring this to our attention.
